_posts/2022-04-07-how-to-use-kerberos-in-copr.md

```diff
@@ -0,0 +1,84 @@
+To work with another (non-Fedora Copr) instance, you need to obtain the kerberos ticket differently, and you need to have a configuration file containing the appropriate copr_url:
+
+```shell
+$ kinit username@EXAMPLE.COM
+$ cat ~/.config/copr
+[copr-cli]
+copr_url = https://copr.example.com/
+```
+
+## API key vs. Kerberos
+If you have an API key set up, and you also have a kerberos ticket, then the API key has higher priority and will be used.
+We enable the use of GSSAPI in CLI by default. It means that if you have properly set up the API key, it will be used but if you don't, then we try to use GSSAPI to obtain the necessary information.
+Be careful, in this case CLI will use the `copr_url` set up in the configuration file to get the information, and if you don't have the configuration file at all, it will use the default url,
+which is `https://copr.fedorainfracloud.org`. If you want to use another instance of Copr, set `copr_url` to the preferred instance. You can disable this behaviour by setting `gssapi` to False in
+the configuration file.
+
+```shell
+[copr-cli]
+gssapi = False
+```
+Unlike cli, this behavior is disabled by default in the API. You can still use the method `create_from_config_file`.
+
+```python
+from copr.v3 import Client
+client = Client.create_from_config_file()
+print(client.config)
+```
+```python
+{'username': None, 'login': None, 'token': None, 'copr_url': 'https://copr.fedorainfracloud.org', 'gssapi': None, 'encrypted': True}
+```
+You can see that `gssapi` is not set so anything you do now using the API and requires authentication will fail and throw an exception
+`CoprAuthException`. You can solve this by setting:
+
+```python
+client.config["gssapi"] = True
+```
+
+## Switch primary ticket cache
+
+If you obtain a ticket e.g. for `username@REDHAT.COM`, and then you obtain a ticket for `username@FEDORAPROJECT.ORG` everything works
+well. In CLI and API it does not matter in which order you get the ticket. However, if you obtain a ticket first for `FEDORAPROJECT.ORG` 
+and then for `REDHAT.COM`, it will not be possible to log in to the WebUI and a message will appear stating that the server 
+could not verify that you are authorized to access the document requested. You can solve this issue by using `kswitch`.
+
+```shell
+kswitch -p username@FEDORAPROJECT.ORG
+```
+it makes the specified credential cache the primary cache for the collection, if a cache collection is available.
+No matter the order in which you get the ticket now when you list cached Kerberos tickets using `klist`
+
+```shell
+klist -A
+```
+you can see that Kerberos ticket for `username@FEDORAPROJECT.ORG` is now listed as first.
+
+## New users
+If you are a new user and want to start using Copr, you will first need to log in to the WebUI using the `log in` link.
+We are currently unable to retrieve groups and email from fedora accounts. Behind the `log in` link is `openid`, which can do this.
```

Comment:
Or if your group membership changed and you want to use this new group in Copr.  We are currently unable to retrieve group membership and your email from FAS when you log in using Kerberos.